ros2_batch_job/__main__.py

```diff
@@ -295,6 +299,13 @@ def build_and_test(args, job):
         cmd.append('--cmake-args')
         cmd.extend(cmake_args)
 
+    if compile_on_clang:
+        install_clang_cmd = ['sudo apt-get install --no-install-recommends -y clang;']
```

Comment:
We've typically done package installation for Linux in the Dockerfiles; see https://github.com/ros2/ci/blob/master/linux_docker_resources/Dockerfile#L35 and the lines below for examples.  Both for consistency and because this file is shared between Linux and the other platforms, I think we should move this package installation over to the Dockerfile, conditional on some variable passed in there.

Reply:
Agreed, I actually tried to make it work but was facing some issue in matching the variable contents to the expected value. Will fix this and update in next revision.

Reply:
Moved the installation of clang to dockerfile and based it on a build arg "COMPILE_WITH_CLANG"